frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,174 @@
+
+  if (isHovered) {
+    return href
+  }
+
+  return undefined
+}
+
+export const linkCellRenderer: CustomRenderer<LinkCell> = {
+  draw: (args, cell) => {
+    const { ctx, rect, theme, hoverX = -100, highlighted } = args
+    const { href, displayText } = cell.data
+    if (href === undefined) return
+
+    const displayValue = displayText || href
+
+    const xPad = theme.cellHorizontalPadding
+
+    const drawX = rect.x + xPad
+
+    const rectHoverX = rect.x + hoverX
+
+    const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+    ctx.font = font
+    const middleCenterBias = getMiddleCenterBias(ctx, font)
+    const drawY = rect.y + rect.height / 2 + middleCenterBias
+
+    const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + rect.width
+
+    if (isHovered) {
+      // draw the underline only when the cell is hovered
+      const metrics = measureTextCached(displayValue, ctx, font)
+
+      ctx.moveTo(drawX, Math.floor(drawY + UNDERLINE_OFFSET) + 0.5)
+      ctx.lineTo(
+        drawX + metrics.width,
+        Math.floor(drawY + UNDERLINE_OFFSET) + 0.5
+      )
+
+      ctx.strokeStyle = theme.linkColor
+      ctx.stroke()
+
+      ctx.fillStyle = highlighted
+        ? blend(theme.accentLight, theme.bgCell)
+        : theme.bgCell
+      ctx.fillText(displayValue, drawX - 1, drawY)
+      ctx.fillText(displayValue, drawX + 1, drawY)
+
+      ctx.fillText(displayValue, drawX - 2, drawY)
+      ctx.fillText(displayValue, drawX + 2, drawY)
+    }
+
+    ctx.fillStyle = theme.linkColor
+    ctx.fillText(displayValue, drawX, drawY)
+    ctx.closePath()
+
+    return true
+  },
+  isMatch: (c): c is LinkCell =>
+    (c.data as LinkCellProps).kind === "link-cell",
+  kind: GridCellKind.Custom,
+  measure: (ctx, cell, theme) => {
+    const { href, displayText } = cell.data
+    if (href === undefined) return 0
+
+    return (
+      ctx.measureText(displayText || href).width +
+      theme.cellHorizontalPadding * 2
+    )
+  },
+  needsHover: true,
+  needsHoverPosition: true,
+  onSelect: e => {
+    const redirectLink = onClickSelect(e)
+
+    if (redirectLink !== undefined) {
+      window.open(redirectLink, "_blank")
+    }
+  },
+  onDelete: c => ({
+    ...c,
+    data: {
+      ...c.data,
+      displayText: "",
+      href: "",
+    },
+  }),
+  provideEditor: () => p => {
+    const { onChange, value, forceEditMode, validatedSelection } = p
+    const { href, displayText } = value.data
+    return (
+      <UriOverlayEditor
+        forceEditMode={forceEditMode}
+        uri={value.data.href}
+        preview={displayText || href}
+        validatedSelection={validatedSelection}
+        readonly={value.readonly === true}
+        onChange={e =>
+          onChange({
+            ...value,
+            copyData: e.target.value,
+            data: {
+              ...value.data,
+              href: e.target.value,
+              displayText: value.data.displayText,
```

Comment:
Since `displayText` cannot be modified with this implementation, we don't need to set it here

```suggestion
```